providers/implementations/macs/kmac_prov.c

```diff
@@ -535,6 +535,9 @@ static int bytepad(unsigned char *out, size_t *out_len,
                    const unsigned char *in1, size_t in1_len,
                    const unsigned char *in2, size_t in2_len, size_t w)
 {
+    if (w == 0)
```

Comment:
A few lines down there is

```
if (!ossl_assert(w <= 255))
    return 0;
```

Would it be a better fix to change the assert to `w > 0 && w <= 255` at move it to before `w` gets used?